worker/sshserver/listener_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"sync"
+
+	"github.com/juju/testing"
+	"google.golang.org/grpc/test/bufconn"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/worker/sshserver"
+)
+
+type listenerSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&listenerSuite{})
+
+func (s *listenerSuite) TestAcceptOnceListener(c *gc.C) {
+	listener := bufconn.Listen(8 * 1024)
+
+	acceptOnceListener, closeAllowed := sshserver.NewAcceptOnceListener(listener)
+	c.Assert(acceptOnceListener, gc.NotNil)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		<-closeAllowed
+		listener.Close()
```

Comment:
actually, i think a better test would be without this goroutine.. so 
```
  done := make(chan bool, 1)

  go func() {
    defer func() {
      done <- true
    }()
    // Accept runs and sends down the channel, it is blocked then until
    // Close continues.
    acceptOnceListener.Accept()
  }()

  <-closeAllowed
  listener.Close()

  select {
      case <-done:
      case <-time.After(somewait):
        c.Fail()
  }
```

Reply:
I preferred has it was, but updated